maven-projects/info/src/main/java/org/apache/graphar/info/loader/BaseGraphInfoLoader.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.graphar.info.loader;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.graphar.info.EdgeInfo;
+import org.apache.graphar.info.GraphInfo;
+import org.apache.graphar.info.VertexInfo;
+import org.apache.graphar.info.yaml.AdjacentListYaml;
+import org.apache.graphar.info.yaml.EdgeYaml;
+import org.apache.graphar.info.yaml.GraphYaml;
+import org.apache.graphar.info.yaml.PropertyGroupYaml;
+import org.apache.graphar.info.yaml.VertexYaml;
+import org.apache.graphar.util.PathUtil;
+
+public abstract class BaseGraphInfoLoader implements GraphInfoLoader {
+
+    public abstract GraphInfo loadGraphInfo(URI graphYamlUri) throws IOException;
+
+    public abstract VertexInfo loadVertexInfo(URI vertexYamlUri) throws IOException;
+
+    public abstract EdgeInfo loadEdgeInfo(URI edgeYamlUri) throws IOException;
+
+    public GraphInfo buildGraphInfoFromGraphYaml(URI baseUri, GraphYaml graphYaml)
+            throws IOException {
+        String prefix = PathUtil.pathToDirectory(baseUri.getPath());
```

Comment:
Why not pass URI to the PathUtil directly? It will allows us to get scheme (`file:///`, `ftp://`, `s3a://`, `hdfs://`, etc.) instead of checking it by `startsWith`. I think relying on the Java built-in API will make it more robust and flexible, compared to raw strings manipulations when we should cover all the corner cases...